src/shared/container-engine-lib.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# container engine helper functions to handle both podman and docker commands
+
+CICD_TOOLS_COMMON_LOADED=${CICD_TOOLS_COMMON_LOADED:-1}
+
+if [ "$CICD_TOOLS_CONTAINER_ENGINE_LOADED" -eq 0 ]; then
+    return 0
+fi
+
+if [ "$CICD_TOOLS_COMMON_LOADED" -ne 0 ]; then
+    source "${CICD_TOOLS_WORKDIR}/src/shared/common.sh"
+fi
+
+echo "loading container engine"
+
+CONTAINER_ENGINE_CMD=''
+
+container_engine_cmd() {
+
+    if [ -z "$(get_container_engine_cmd)" ]; then
+        if ! set_container_engine_cmd; then
+            return 1
+        fi
+    fi
+
+    if [ "$(get_container_engine_cmd)" = "podman" ]; then
+        podman "$@"
+    else
+        docker "--config=${DOCKER_CONF}" "$@"
```

Comment:
We should not need to run docker with `--config` if the `DOCKER_CONFIG` env var is set, and the other benefit is both podman and docker honor that env var.

Reply:
This section has been refactor to remove adding the `--config` bit and now it's simpler. thanks!